rfp-proposal/identity-directory.md

```diff
@@ -0,0 +1,104 @@
+
+It accepts input such as https://kusama.polkaperson.com/swader or https://kusama.polkaperson.com/HviHUSkM5SknXzYuPCSfst3CXK4Yg6SWeroP6TdTZBZJbVT or https://polkaperson.com/ThJ.
+
+The service:
+
+- reads the input param
+- if the input param is an address or an index (second and third option respectively), the [single page view](#single-page-view) opens
+- if the input param is not an address (first option), a [list page view](#list-page-view) opens
+
+### List Page View
+
+The service looks through all the registered identities on Kusama or Polkadot, and lists all the identities that have the provided value set as ANY of the fields in their identity entry. Clicking any of them takes a person to the [Single Page View](#single-page-view) and changes the URL to https://polkaperson.com/HviHUSkM5SknXzYuPCSfst3CXK4Yg6SWeroP6TdTZBZJbVT with the appropriate address, since addresses are unique.
+
+Alternatively, if the address has a [**frozen index**](https://wiki.polkadot.network/docs/en/learn-accounts#indices) attached, the index should replace the address in the URL for usability, i.e. https://polkaperson.com/ThJ.
+
+### Single Page View
+
+The single page view contains a beautifully rendered identity card of an on-chain identity along with all the metadata in the identity entry, an avatar if provided, any verifications from registrars in the chain, and buttons somewhere in the UI that allow you to Send tokens directly to the user. 
+
+![](https://i.imgur.com/PpQy855.jpg)
+
+*Above screenshot is from [Pulse HTML template](https://themeforest.net/item/pulse-personal-academic-vcard-template/10654629).*
+
+The UI should be tweetdeck-like in that it offers multiple columns, each with a specific purpose, and the columns should be close-able and re-orderable, so users can just order them as they choose. The order should be remembered across the app, and there should be a toggle to **freeze** a layout for a specific user (i.e. maybe you only want Validator Performance for a certain validator but maybe you're interested in *everything* a high-profile account does on chain so you want all the columns).
+
+### Plugins
+
+The application should support a plug-in ecosystem for different sub-views of identities. The hosted, on-line version would come with some default plugins activated (see [Default Plugins](#default-plugins)) while the off-line version should support simple installation of other plugins. Most plug-ins will be rendered as **additional columns of data**, except in some cases (i.e. replace the "recent news" ticket at the top of the screen with a "recent on-chain tips issued", or a plug-in which adds push notifications).
+
+Note: **if a user is not connecting to an archive node, archive data should be unavailable**. The plugins (default, optional, and third party plug-ins) are responsible for realizing this and reporting to the user that some info won't be available if the user connects to only a full node.
```

Comment:
Is an archive node enough, or would an indexer be necessary (= at least for usable performance) for some plugins?

Reply:
It depends on the plugin and how it's built. For the default things, an archive node should be enough. Other plugins can be developed to be completely centralized, it's up to the user to pick how usable vs how decentralized they want their layout to be.